text/0000-native-link-modifiers.md

```diff
@@ -0,0 +1,232 @@
+and currently there's no way to apply them.
+
+For example, some static libraries may need to be linked as a whole archive
+without throwing any object files away because some symbols in them appear to be unused
+while actually having some effect. \
+This RFC introduces modifier `whole-archive` to address this.
+
+In other cases we need to link to a library located at some specific path
+or not matching the default naming conventions. \
+This RFC introduces modifier `verbatim` to pass such libraries to the linker.
+
+This RFC also reformulates the `static-nobundle` linking kind as a modifier `bundle`
+thus providing an opportunity to change the static linking default to non-bundling
+on some future edition boundary, and hopefully unblocking its stabilization.
+
+The generic syntax provides a way to add more such modifiers in the future
+without introducing new linking kinds.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This is an advanced feature not expected to be used commonly,
+see the reference-level explanation if you know that you need some of these modifiers.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Existing syntax of linking attributes and options
+
+- Attributes: `#[link(name = "string", kind = "string", cfg(predicate))]`
+(some components are optional.)
+- Command line options: `-l kind=name:rename` (some components are optional).
+
+## Proposed extensions to the syntax
+
+- Attributes: `#[link(/* same */, modifiers = "+foo,-bar")]`.
+- Command line options: `-l kind:+foo,-bar=name:rename`.
+
+The modifiers are boolean and applied only to the single library specified with `name`. \
+`+` means enable, `-` means disable, multiple options can be separated by commas,
+the last boolean value specified for the given modifier wins. \
+The notation is borrowed from
+[target features](https://doc.rust-lang.org/rustc/codegen-options/index.html#target-feature)
+in general and should have the same semantics.
+
+If the `:rename` component is specified on the command line, then in addition to the name
+and linking kind the modifiers will be updated as well (using concatenation).
+
+## Specific modifiers
+
+### `bundle`
+
+Only compatible with the `static` linking kind.
+
+`+bundle` means objects from the static library are bundled into the produced crate
+(a rlib, for example) and are used from this crate later during linking of the final binary.
+
+`-bundle` means the static library is included into the produced rlib "by name"
+and object files from it are included only during linking of the final binary,
+the file search by that name is also performed during final linking.
+
+This modifier is supposed to supersede the `static-nobundle` linking kind defined by
+[RFC 1717](https://github.com/rust-lang/rfcs/pull/1717).
+
+The default for this modifier is currently `+bundle`,
+but it could be changed later on some future edition boundary.
```

Comment:
If the whole point of editions is that you can link together code for multiple editions at once, wouldn't that not be a good idea?

Reply:
I don't think this change would break anything implicitly?
When a crate with `#[link]` attributes migrates from edition 202X to edition 202X+3 it will need to either add the modifier or change its build and/or distribution strategy. Both such such changes should be ok when a crate does edition bump.